### DISCUSSION TITLE
Best way to transmit several FSK packets?

### DISCUSSION THREAD

**Discussion by tauesdospi** [jgromes/RadioLib#919] [General]
Hi
I'm trying to send multiple consecutive FSK packets in SX1262 without luck (ESP32 riscv Heltec minishell module)
I have a state machine inside the arduino loop which chops the a big buffer into chunks of eg. 240 bytes and I'm transmitting via radio.transmit() and wait up to radio.getTimeOnAir(). I have also check with interrupt mode (when a flag is set, then the packet tx is complete and i send another chunk).

First question is what is the best mode to achive this?

At the reception side I'm using radio.startReceive and after the interrupt tells the packet is ready I read it via radio.getPacketLength,+radio.readData.  

I'm getting inconsistent errors and the last packet of a group of packets is not received all the times (eg. if the 1st fsk packet is 240 bytes and the 2nd packet is < 200 bytes, then the 2nd packet is not received)? (I have the FSK packet size at up to 240 bytes each). Variable length mode is enabled.

Can you provide any light on this?

Thanks you very much!
Antonio

**Comment by jgromes** [comment]
Transmitting multiple FSK packets should be as straight-forward as calling `radio.transmit()` multiple times, so you understood that correctly.

Of course, you need to make sure that the receiver has enough time to receive and process everything. This may be longer than what `getTimeOnAir` returns, because that just tells you how long will the packet transmission take, it cannot take into account demodulation, decoding and processing on the receiving end.

So the overall logic looks almost good, but I would have to see the actual code to comment any further.

**Reply by tauesdospi** [reply]
Found the problem: I'm debugging the process via UART print. It takes more time to print that the time of arrival of a new packet and I got some overwrite issues..
Thanks for your continued support!
Regards
Antonio
